### PR TITLE
Upgrade stack to lts-18.19 and use Stack on CI 🆕

### PR DESCRIPTION
**Pull request by kutyel**
Keeping things up to date @serras 😉🧹

### CODE REVIEW

File: .github/workflows/haskell.yml [hunk 1]
```diff
@@ -1,29 +1,29 @@
...
-on: [push]
+on: [push, pull_request]
 jobs:
   build:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        ghc: ["8.8.3"]
-        cabal: ["3.0"]
+        ghc: ["8.8.3", "9.2.1"]
```
Comment (serras) [inline .github/workflows/haskell.yml]: I would suggest to keep 8.10.7 and 9.2.1, and use only one Cabal

Comment (kutyel) [reply .github/workflows/haskell.yml]: build failed telling me it could not find 8.10.7, maybe the action uses an outdated version of `ghcup`? 🤔

Comment (serras) [reply .github/workflows/haskell.yml]: According to the docs, you should migrate to https://github.com/haskell/actions/tree/main/setup, the other one is no longer maintained

**Review by serras — CHANGES_REQUESTED** [review]
Please update to the new version of https://github.com/haskell/actions/tree/main/setup, that way we can test with recent versions of GHC

**Review by serras — APPROVED** [review]

**Pull request merged**